### PR TITLE
lazy loading dependencies

### PR DESCRIPTION
**Pull request by kobik**

### CODE REVIEW

File: src/utils/schemaValidators.js [hunk 1]
```diff
@@ -1,10 +1,21 @@
-const OpenAPISchemaValidator = require('openapi-schema-validator').default;
-const openApi3Validator = new OpenAPISchemaValidator({
-    version: 3,
-    version3Extensions: {
-    }
-});
+
+const getOAI3Validator = (() => {
```
Comment (kibertoad) [inline src/utils/schemaValidators.js:2]: why not just make it a named function?

File: src/utils/schemaValidators.js [hunk 2]
```diff
@@ -1,10 +1,21 @@
...
+    const OpenAPISchemaValidator = require('openapi-schema-validator').default;
+
+    let validator;
```
Comment (kibertoad) [inline src/utils/schemaValidators.js:5]: Why do this as a function within function and not just have top-level variable `validator`?

File: src/utils/schemaValidators.js (hunk 1 above)
Comment (kobik) [reply src/utils/schemaValidators.js]: you're right, i had something else on mind.

File: src/utils/fileLoaders.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
+const fs = require('fs');
+const path = require('path');
+const jsyaml = require('js-yaml');
+
+const cwd = process.cwd();
+
+const file = (refValue, options) => {
```
Comment (kibertoad) [inline src/utils/fileLoaders.js:7]: Typical convention is for functions to start with verbs. Currently it is really unclear what it does.

File: src/utils/fileLoaders.js [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    let refPath = refValue;
```
Comment (kibertoad) [inline src/utils/fileLoaders.js:8]: is this needed? why not just do immutable assignment in a ternary statement below?

File: src/utils/fileLoaders.js [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+    if (refPath.indexOf('file:') === 0) {
+        refPath = refPath.substring(5);
+    } else {
+        refPath = path.resolve(baseFolder, refPath);
+    }
+
+    const filePath = getRefFilePath(refPath);
+    const filePathLowerCase = filePath.toLowerCase();
+
+    var data = fs.readFileSync(filePath, 'utf8');
```
Comment (kibertoad) [inline src/utils/fileLoaders.js:20]: why var?

File: src/utils/fileLoaders.js [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+    if (filePathLowerCase.endsWith('.json')) {
+        return JSON.parse(data);
+    } else if (filePathLowerCase.endsWith('.yml') || filePathLowerCase.endsWith('.yaml')) {
+        return jsyaml.load(data);
+    }
+};
+
+function getRefFilePath(refPath) {
+    let filePath = refPath;
```
Comment (kibertoad) [inline src/utils/fileLoaders.js:29]: this assignment seems redundant, probably you can simply do early return instead

File: src/utils/fileLoaders.js (hunk 3 above)
Comment (kobik) [reply src/utils/fileLoaders.js:7]: this what `json-schema-deref-sync` requires in loaders

File: src/utils/fileLoaders.js (hunk 6 above)
Comment (kobik) [reply src/utils/fileLoaders.js]: changed

File: src/utils/fileLoaders.js (hunk 5 above)
Comment (kobik) [reply src/utils/fileLoaders.js]: don't know... fixed

File: src/utils/fileLoaders.js (hunk 4 above)
Comment (kobik) [reply src/utils/fileLoaders.js:8]: changed code a bit

**Comment by kibertoad** [comment]
@kobik According to https://github.com/PayU/openapi-validator-middleware/issues/116#issuecomment-653123276, this provides 40% startup time reduction when not using OpenAPI 3.0, so this seems to be clearly helping. Shall we merge it then?

**Pull request merged**